cypress-io/cypress
# feat(unify): toggle specs list

Closes https://cypress-io.atlassian.net/browse/UNIFY-551

### User facing changelog

The inline specs list in the runner can be opened/closed by means of the Specs button in the reporter, as well as the shortcut key `f`.

### Additional details

This PR does some other things (paired with @lmiller1990 on some of this):

- uses the Runner UI Store for Vue state of the specs list as open/closed, but also to initialize the reporter with the correct value from gql once the localPreferences have been loaded
- fixes a missing hookup for the `autoScrollingEnabled` flag using the same flow as the `isSpecsListOpen` state
- fixes a tiny bug with saving `autoScrollingEnabled` when toggled via keyboard shortcut
- there's some CSS stuff from @mapsandapps that seems like it's fine to stay in - Shawn let me know if this is no longer needed?

### How has the user experience changed?

https://user-images.githubusercontent.com/8340719/144281985-10a0a97b-934d-45e8-9d1e-59a74b3b58f1.mov

### How to test

- Run `yarn dev` for a project with specs and go to a spec.
- Toggle specs list with `f` key (when not focused in a text field or the AUT)
- Toggle specs list with the Specs button in the reporter
- Search for something in the specs list and toggle - see that specs list internal state is maintained
- See that open/closed state is maintained after page refresh, navigation, and full stop and restart of Cypress

### PR Tasks


- [na] Has the original issue or this PR been tagged with a release in ZenHub? 
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?
